confiture-rest-api/src/audits/audit.service.ts

```diff
@@ -183,8 +185,8 @@ export class AuditService {
     data: UpdateAuditDto
   ): Promise<Audit | undefined> {
     try {
-      const updatedPages = data.pages.filter((p) => p.id);
-      const newPages = data.pages.filter((p) => !p.id);
+      const updatedPages = data.pages.sort((p) => p.order).filter((p) => p.id);
+      const newPages = data.pages.sort((p) => p.order).filter((p) => !p.id);
```

Comment:
Il faut utiliser l'index qu'ont les pages dans la liste originelle `data.pages`. Sinon, on se retrouve avec des pages qui ont même le `order`


```suggestion
      const orderedPages = data.pages.map((p, i) => ({ ...p, order: i }));
      const updatedPages = orderedPages.filter((p) => p.id);
      const newPages = orderedPages.filter((p) => !p.id);
```